cluster-up/cluster/k8s-provider-common.sh

```diff
@@ -1,9 +1,22 @@
 #!/usr/bin/env bash
 
-set -e
+set -eo pipefail
```

Comment:
This file is sourced, so I don't think it is a good idea to set this here. I'd rather prefer this set in the calling scripts.

Reply:
Although this is done in other places several times already? WDYT?

Reply:
Let's run provision tests (they run cluster-up too) so we see if we good ?

Reply:
you can also if you want use just to that command
set -o pipefail && command

btw this phenomena will potentially hit us on more places, silently
i saw it happens when i created 3 masters PR  , where it happens a lot